src/reducers/nft/index.js

```diff
@@ -47,6 +55,31 @@ async function getCachedContractMetadataOrFetch(contractName, state) {
     return getMetadata(contractName);
 }
 
+const fetchNFTsByContractName = createAsyncThunk(
```

Comment:
1. I think we can use a single action to handle inserting token metadata into the store, for both the 'initial load' and 'adding a new page of tokens' cases, as long as we ensure that it handles creating an array for a particular contract if none exists, rather than have two different actions that are slightly different :).  It seems like `addTokensMetadata` reducer actually might do that already, which makes `setTokensMetadata` unnecessary :). Note that we clear the existing token state on account change, so as long as the reducer can create the array in state when it receives a `tokens` array we're good to go on just using `addTokensMetadata` and can delete the existing action entirely.

2. This function is almost an exact copy of the code inside of `fetchNFTs` `Promise.all()` map.  Given that (1) above is done, then I think we can dispatch _this_ thunk from inside the Promise.all() callback in `fetchNFTs` rather than duplicate the code in both places, with two details:
a. Make sure that `selectOwnedTokensForAccountForContract()` returns an empty array if it doesn't find any existing tokens so that the .length check on L72 returns 0 :)
b. This function should not be squashing errors, because it is going to be called based on a direct user interaction and we will want to show an error to the user if we failed to load more NFTs for a contract that we'd previously interacted with successfully (which is _not_ the case in `fetchNFTs`, where as an initial load we have no confidence that the likelyContracts is going to return a list of fully NEP141 compatible contract names).  So the `catch()` block here on L76/77 should be removed so that the caller can catch errors loading tokens when someone clicks 'load more' and it fails.  When you have updated `fetchNFTs` to call this thunk inside the promise.all() callback below, errors thrown from this thunk will still be caught during initial `fetchNFTs()`  execution due to the catch on L105, so we don't have to worry about it bubbling up to the console during initial fetch of contract during page load or after an account switch.

Reply:
Done.